internal/dockercompose/dockercompose.go

```diff
@@ -1,85 +1,165 @@
 package dockercompose
 
 import (
-	"gopkg.in/yaml.v3"
+	"bytes"
+	"fmt"
 	"os"
+	"runtime"
+
+	"github.com/dchest/uniuri"
+	"gopkg.in/yaml.v3"
+)
+
+type OS int
+
+const (
+	WindowsMac OS = iota
+	Linux
 )
 
-func CreateDockerComposeFile(dest, latestVersion, creDir string, noCloud bool) error {
-	destFullPath := os.Getenv("PWD") + "/" + dest
+func CreateDockerComposeFile(gitopsPath, gitopsName, latestGitopsVersion, latestBitswanEditorVersion, certsPath string) (string, error) {
 	sshDir := os.Getenv("HOME") + "/.ssh"
-	if creDir == "" {
-		creDir = "cre-01"
+	gitConfig := os.Getenv("HOME") + "/.gitconfig"
+
+	hostOsTmp := runtime.GOOS
+
+	var hostOs OS
+	switch hostOsTmp {
+	case "windows", "darwin":
+		hostOs = WindowsMac
+	case "linux":
+		hostOs = Linux
+	default:
+		return "", fmt.Errorf("unsupported host OS: %s", hostOsTmp)
+	}
+
+	// generate a random secret token
+	gitopsSecretToken := uniuri.NewLen(64)
+
+	gitopsService := map[string]interface{}{
+		"image": "bitswan/gitops:" + latestGitopsVersion,
+		"restart": "always",
+		"networks": []string{"bitswan_network"},
+		"volumes": []string{
+			"/etc/bitswan-secrets/:/etc/bitswan-secrets/", // TODO: change this to gitops secrets
```

Comment:
Isn't this outdated?

Reply:
Yeah I was waiting on the gitops changes regarding this so I can properly mount the secrets and set the env var...I forgot about it..are changes in gitops regarding this already pushed?